trace-utils/src/span/v05/mod.rs

```diff
@@ -29,24 +32,146 @@ pub struct Span {
+        state.serialize_field("time_unix_nano", &self.0.time_unix_nano)?;
+        state.serialize_field("name", &self.0.name)?;
+        state.serialize_field(
+            "attributes",
+            &SpanEventAttributesSerializerV05(&self.0.attributes),
+        )?;
+        state.end()
+    }
+}
+
+impl serde::Serialize for SpanEventAttributesSerializerV05<'_> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::ser::Serializer,
+    {
+        use serde::ser::SerializeMap;
+        let mut map = serializer.serialize_map(Some(self.0.len()))?;
+        for (key, value) in self.0 {
+            map.serialize_entry(key, &AttributeAnyValueV05(value))?;
+        }
+        map.end()
+    }
+}
+
+impl serde::Serialize for AttributeAnyValueV05<'_> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::ser::Serializer,
+    {
+        use serde::ser::SerializeSeq;
+        match self.0 {
+            AttributeAnyValueBytes::SingleValue(v) => {
+                AttributeArrayValueV05::from_inner(v).serialize(serializer)
+            }
+            super::AttributeAnyValue::Array(attribute_array_values) => {
+                let mut seq = serializer.serialize_seq(Some(attribute_array_values.len()))?;
+                for value in attribute_array_values {
+                    seq.serialize_element(&AttributeArrayValueV05::from_inner(value))?;
+                }
+                seq.end()
+            }
+        }
+    }
+}
+
+#[derive(Debug, Serialize)]
+#[serde(untagged)]
+pub enum AttributeArrayValueV05<'a> {
+    String(&'a BytesString),
+    Boolean(bool),
+    Integer(i64),
+    Double(f64),
+}
+
+impl<'a> AttributeArrayValueV05<'a> {
+    fn from_inner(v: &'a AttributeArrayValueBytes) -> Self {
+        use AttributeArrayValue::*;
+        match v {
+            String(string) => AttributeArrayValueV05::String(string),
+            Boolean(boolean) => AttributeArrayValueV05::Boolean(*boolean),
+            Integer(integer) => AttributeArrayValueV05::Integer(*integer),
+            Double(double) => AttributeArrayValueV05::Double(*double),
+        }
+    }
+}
+
 pub fn from_span_bytes(span: &SpanBytes, dict: &mut SharedDict) -> Result<Span> {
+    let service = dict.get_or_insert(&span.service)?;
+    let name = dict.get_or_insert(&span.name)?;
+    let resource = dict.get_or_insert(&span.resource)?;
+    let mut meta = span.meta.iter().try_fold(
+        HashMap::with_capacity(span.meta.len()),
+        |mut meta, (k, v)| -> anyhow::Result<HashMap<u32, u32>> {
+            meta.insert(dict.get_or_insert(k)?, dict.get_or_insert(v)?);
+            Ok(meta)
+        },
+    )?;
+    if !span.span_links.is_empty() {
+        let serialized_span_links = serde_json::to_string(&span.span_links)?;
+        meta.insert(
+            dict.get_or_insert(&tinybytes::BytesString::from("span_links"))?,
```

Comment:
In the RFC there is some discussion about "span_links" or "_dd.span_links". Did they finally choose the former?

Reply:
No, it's a mistake the key should definitely be "_dd.span_links" from other implementations